AISKU/scripts/setActiveCdnVersion.ps1

```diff
@@ -0,0 +1,668 @@
+
+            Log $("  - {0,-40} {1,6:N1} Kb  {2:yyyy-MM-dd HH:mm:ss}" -f $($blob.ICloudBlob.Container.Name + "/" + $blob.Name),($blob.Length/1kb),$blob.LastModified)
+            $theBlob = [hashtable]@{}
+            $theBlob.blob = $blob
+            $theBlob.context = $context
+            $fileList.Add($theBlob)
+        }
+    }
+}
+
+Function RemoveMetadata(
+    $cloudBlob,
+    [string] $dataKey
+) {
+    # Removing and adding the attribute to avoid duplication of values when the key case is different
+    $changed = $true
+    while ($changed -eq $true) {
+        $changed = $false
+        foreach ($dstKey in $cloudBlob.Metadata.Keys) {
+            if ($dstKey -ieq $dataKey) {
+                $cloudBlob.Metadata.Remove($dstKey) | Out-Null
+                $changed = $true
+                break
+            }
+        }
+    } 
+}
+
+Function CopyBlob(
+    $blobContext,
+    $blob,
+    $destContext,
+    $destName
+) {
+    Log-Errors
+
+    # Don't perform any copyies if if any errors have been logged as we want to make sure the attributes have been set
+    if ($global:hasErrors -eq $true) {
+        exit 2
+    }
+
+    Log "       - $($blob.Name) ==> $destName"
+
+    $srcCloudBlob = $blob.ICloudBlob.FetchAttributes()
+
+    $blobResult = Start-AzureStorageBlobCopy -Context $blobContext -CloudBlob $blob.ICloudBlob -DestContext $destContext.azureContext -DestContainer "$($destContext.storageContainer)" -DestBlob $destName -Force
+    Log-Errors
+
+    # Don't try and publish anything if any errors have been logged
+    if ($global:hasErrors -eq $true) {
+        exit 2
+    }
+    
+    $status = $blobResult | Get-AzureStorageBlobCopyState
+    while ($status.Status -eq "Pending") {
+        $status = $blobResult | Get-AzureStorageBlobCopyState
+        Log $status
+        Start-Sleep 10
+    }
+
+    # Don't try and publish anything if any errors have been logged
+    if ($global:hasErrors -eq $true) {
+        exit 2
+    }
+
+    # Make sure the metadata and properties are set correctly
+    # - When destination did not exist then the properties and metadata are set correctly
+    # - But when overwriting an existing blob the properties and metadata are not updated
+    $newBlob = Get-AzureStorageBlob -Context $destContext.azureContext -Container "$($destContext.storageContainer)" -Blob $destName
+    $cloudBlob = $newBlob.ICloudBlob
+    $cloudBlob.FetchAttributes()
+    $cloudBlob.Properties.CacheControl = $blob.ICloudBlob.Properties.CacheControl
+    foreach ($dataKey in $blob.ICloudBlob.Metadata.Keys) {
+        RemoveMetadata $cloudBlob $dataKey
+        $cloudBlob.Metadata.Add($dataKey, $blob.ICloudBlob.Metadata[$dataKey]) | Out-Null
+    }
+
+    $cloudBlob.SetProperties()
+    $cloudBlob.SetMetadata()
+}
+
+Function SetProperties(
+    $stagedBlob,
+    $srcName,
+    $ver
+) {
+    $cloudBlob = $stagedBlob.ICloudBlob
+    $cloudBlob.FetchAttributes()
+    $cloudBlob.Properties.CacheControl = $cacheControl30Min
```

Comment:
Would it make sense for this to be configurable? I think it would be cool if this were set up in a multi phase deployment with 30min for initial release, and then 48?hours later is redeployed with X hours. WDYT?

Reply:
We don't really want to make this value too big as it will take that long for the next release to get picked up, we could have a smaller rate (15min then 30min perhaps), but the downside is that every browser will potentially try to fetch the script again during that interval

I have another work item to minimize the size impacts so that the scripts are cached longer and/or the initial script is not so large.